### PR TITLE
[IOS-465]: Support empty group avatar

### PR DESCRIPTION
**Pull request by yuliaveres**
Get rid of not needed assertions related to empty/bad data in attachments.

Needed to have group chat working properly when a bot can be a group member; since a bot can send to group whatever, or  create a group without avatar f.e.

### CODE REVIEW

File: Pods/SignalServiceKit/SignalServiceKit/src/Messages/OWSMessageSender.m [hunk 1]
```diff
@@ -918,7 +918,7 @@ - (void)sendMessage:(TSOutgoingMessage *)message
 
             NSData *newIdentityKeyWithVersion = newKeyBundle.identityKey;
 
-            if (![newIdentityKeyWithVersion isKindOfClass:[NSData class]]) {
+            if (newIdentityKey && ![newIdentityKeyWithVersion isKindOfClass:[NSData class]]) {
```
Comment (designatednerd) [inline Pods/SignalServiceKit/SignalServiceKit/src/Messages/OWSMessageSender.m]: out of curiosity, why is this necessary?

File: Pods/SignalServiceKit/SignalServiceKit/src/Messages/TSGroupModel.h [hunk 2]
```diff
@@ -20,19 +22,21 @@ extern NSString *const GroupMemberJoinedMessage;
 
 @property (nonatomic, strong) NSArray<NSString *> *groupMemberIds;
 @property (nonatomic, strong) NSString *groupName;
-@property (nonatomic, strong) NSData *groupId;
+@property (nonatomic, strong, nullable) NSData *groupId;
```
Comment (designatednerd) [inline Pods/SignalServiceKit/SignalServiceKit/src/Messages/TSGroupModel.h]: why should the GroupID be nullable?

File: Toshi/Controllers/Chat/ChatInteractor.swift [hunk 3]
```diff
@@ -470,7 +470,8 @@ final class ChatInteractor: NSObject {
             }
 
             let interactor = ChatInteractor(output: nil, thread: thread)
-            interactor.send(image: newGroupModel.groupImage, in: outgoingMessage, completion: completion)
+            let image = newGroupModel.groupImage ?? UIImage(named: "avatar-placeholder")!
```
Comment (designatednerd) [inline Toshi/Controllers/Chat/ChatInteractor.swift]: can you use an `#imageLiteral` instead - that way the compiler whine if the image is removed instead of crashing at runtime.

File: Toshi/Controllers/GroupChat/GroupViewModelProtocol.swift [hunk 4]
```diff
@@ -27,7 +27,7 @@ enum GroupItemType: Int {
 struct GroupInfo {
     let placeholder = Localized("new_group_title")
     var title: String = ""
-    var avatar = UIImage(named: "avatar-placeholder")!
+    var avatar: UIImage? = UIImage(named: "avatar-placeholder")
```
Comment (designatednerd) [inline Toshi/Controllers/GroupChat/GroupViewModelProtocol.swift]: Why does this need to have a nullable avatar? In theory, if the avatar is null, shouldn't we always use the placeholder?

File: Toshi/Controllers/GroupChat/NewGroup/NewGroupViewModel.swift [hunk 5]
```diff
@@ -71,7 +71,9 @@ final class NewGroupViewModel {
 
         completeActionDelegate?.groupViewModelDidStartCreateOrUpdate()
 
-        ChatInteractor.createGroup(with: NSMutableArray(array: groupParticipantsIds), name: groupInfo.title, avatar: groupInfo.avatar, completion: { [weak self] _ in
+        let image = groupInfo.avatar ?? UIImage(named: "avatar-placeholder")!
```
Comment (designatednerd) [inline Toshi/Controllers/GroupChat/NewGroup/NewGroupViewModel.swift]: might not be a bad idea to have an `avatarOrPlaceholder` extension method on the group info so you don't have to do this check every damned time

File: Pods/SignalServiceKit/SignalServiceKit/src/Messages/OWSMessageSender.m (hunk 1 above)
Comment (yuliaveres) [reply Pods/SignalServiceKit/SignalServiceKit/src/Messages/OWSMessageSender.m]: sorry, that was my mistake, fixed already

File: Pods/SignalServiceKit/SignalServiceKit/src/Messages/TSGroupModel.h (hunk 2 above)
Comment (yuliaveres) [reply Pods/SignalServiceKit/SignalServiceKit/src/Messages/TSGroupModel.h]: the id is nil until the group is actually created. (it was always nullable)

File: Toshi/Controllers/GroupChat/GroupViewModelProtocol.swift (hunk 4 above)
Comment (yuliaveres) [reply Toshi/Controllers/GroupChat/GroupViewModelProtocol.swift]: right, it is odd here

File: Toshi/Controllers/GroupChat/GroupInfo/GroupInfoViewModel.swift [hunk 6]
```diff
@@ -88,7 +88,7 @@ final class GroupInfoViewModel {
     private var _sortedMembers: [TokenUser] = []
 
     @objc private func updateGroup() {
-        let image = groupInfo.avatar ?? UIImage(named: "avatar-placeholder")!
+        let image = groupInfo.avatar ?? #imageLiteral(resourceName: "avatar-placeholder")
```
Comment (designatednerd) [inline Toshi/Controllers/GroupChat/GroupInfo/GroupInfoViewModel.swift]: `avatarOrPlaceholder`?

File: Toshi/Controllers/GroupChat/NewGroup/NewGroupViewModel.swift [hunk 7]
```diff
@@ -71,7 +71,7 @@ final class NewGroupViewModel {
 
         completeActionDelegate?.groupViewModelDidStartCreateOrUpdate()
 
-        let image = groupInfo.avatar ?? UIImage(named: "avatar-placeholder")!
+        let image = groupInfo.avatar ?? #imageLiteral(resourceName: "avatar-placeholder")
```
Comment (designatednerd) [inline Toshi/Controllers/GroupChat/NewGroup/NewGroupViewModel.swift]: `avatarOrPlaceholder`?

File: Toshi/Controllers/MessageRequests/MessageRequestsViewController.swift [hunk 8]
```diff
@@ -86,7 +86,7 @@ extension MessagesRequestsViewController: UITableViewDataSource {
 
         guard let thread = dataSource.unacceptedThread(at: indexPath) else { return UITableViewCell(frame: .zero) }
 
-        let avatar = thread.avatar() ?? UIImage(named: "avatar-placeholder")
+        let avatar = thread.avatar() ?? #imageLiteral(resourceName: "avatar-placeholder")
```
Comment (designatednerd) [inline Toshi/Controllers/MessageRequests/MessageRequestsViewController.swift]: `avatarOrPlaceholder`?

File: Toshi/Extensions/TSGroupModel+Additions.swift [hunk 9]
```diff
@@ -13,19 +13,35 @@
...
 
     // true if all data required to display the group is loaded, false if not.
     var isFullyLoaded: Bool {
+        guard let membersIds = groupMemberIds else {
+            return false
+        }
+
         guard
-            (groupMemberIds.count ?? 0) > 1, // There must be more than just the current user as a member of the group
+            (membersIds.count ?? 0) > 1, // There must be more than just the current user as a member of the group
```
Comment (designatednerd) [inline Toshi/Extensions/TSGroupModel+Additions.swift]: in theory the unwrap you do above should remove the need for the `??` section

File: Toshi/Controllers/GroupChat/NewGroup/NewGroupViewModel.swift [hunk 10]
```diff
@@ -71,7 +71,9 @@ final class NewGroupViewModel {
 
         completeActionDelegate?.groupViewModelDidStartCreateOrUpdate()
 
-        ChatInteractor.createGroup(with: NSMutableArray(array: groupParticipantsIds), name: groupInfo.title, avatar: groupInfo.avatar, completion: { [weak self] _ in
+        let image = groupInfo.avatar ?? #imageLiteral(resourceName: "avatar-placeholder")
```
Comment (designatednerd) [inline Toshi/Controllers/GroupChat/NewGroup/NewGroupViewModel.swift]: `avatarOrPlaceholder`?

File: Toshi/Controllers/GroupChat/GroupInfo/GroupInfoViewModel.swift [hunk 11]
```diff
@@ -89,7 +88,8 @@ final class GroupInfoViewModel {
     private var _sortedMembers: [TokenUser] = []
 
     @objc private func updateGroup() {
-        guard let updatedGroupModel = TSGroupModel(title: groupInfo.title, memberIds: NSMutableArray(array: groupInfo.participantsIDs), image: groupInfo.avatar, groupId: thread.groupModel.groupId) else { return }
+        let image = groupInfo.avatar ?? #imageLiteral(resourceName: "avatar-placeholder")
```
Comment (designatednerd) [inline Toshi/Controllers/GroupChat/GroupInfo/GroupInfoViewModel.swift]: `avatarOrPlaceholder`?
